### PR TITLE
fix(events-targets): circular dependency when adding a KMS-encrypted SQS queue 

### LINKED ISSUE #11158: [sqs] Circular dependency between resources when encrypting sqs queue with kms and assigning target to rule
I am setting up an event bridge rule that will forward events from Macie to sqs. Everything works fine until I add `encryptionMasterKey` to the queue. I either have to remove the `.encryptionMasterKey` or remove the `.target()` to avoid circular dependency error.

### Reproduction Steps

   ```
   val queue = Queue.Builder.create(scope, "MacieReportQueue")
                .queueName("MacieReportQueue")
                // TODO: FIXME: KMS Circular dependency error when combined with target
                //  https://github.com/aws/aws-cdk/issues/3067
                .encryption(QueueEncryption.KMS)
                .encryptionMasterKey(key)
                .build()

        val target = SqsQueue.Builder.create(queue).build()

        val rule = Rule.Builder.create(scope, "MacieToSqsEventBridgeRule")
                .ruleName("MacieToSqsEventBridgeRule")
                .description("Sends messages to SQS if Macie finds sensitive information")
                .targets(listOf(target))
                .eventPattern(EventPattern.builder()
                        .source(listOf("aws.macie"))
                        .build()
                )
                .build()
```

[CF Graph](https://ibb.co/JqQq8J6)
### What did you expect to happen?

No circular dependency error. Queue should be encrypted.

### What actually happened?

```
Circular dependency between resources: [MacieReportQueue7D1A954B, MacieReportQueuePolicy84D89C73, MacieToSqsEventBridgeRule498CF606, MacieReportQueueKeyAliasA0A0C899, MacieReportQueueKeyF608AAD8]
```

### Environment

  - **CLI Version      :*1.67.0 (build 2b4dd71)*
  - **Framework Version:*1.67.0*
  - **Node.js Version:**  v14.13.1
  - **OS               :**  Mac 10.14.6
  - **Language (Version):**  Java 8

### Other

--- 

This is :bug: Bug Report

### PR DESCRIPTION
**Pull request by madeline-k**
fixes #11158

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by njlynch** [comment]
@madeline-k - ACK that this will impact both the queue and key policies. I personally think that's acceptable, although I'm open to contrary arguments. I'd proceed with this implementation.

File: packages/@aws-cdk/aws-events-targets/lib/sqs.ts [hunk 1]
```diff
@@ -52,14 +52,18 @@ export class SqsQueue implements events.IRuleTarget {
    * @see https://docs.aws.amazon.com/eventbridge/latest/userguide/resource-based-policies-eventbridge.html#sqs-permissions
    */
   public bind(rule: events.IRule, _id?: string): events.RuleTargetConfig {
-    // deduplicated automatically
-    this.queue.grantSendMessages(new iam.ServicePrincipal('events.amazonaws.com',
-      {
+    // Only add the rule as a condition if the queue is not encrypted, to avoid circular dependency. See issue #11158.
+    var servicePrincipalOpts:iam.ServicePrincipalOpts = {};
```
Comment (njlynch) [inline packages/@aws-cdk/aws-events-targets/lib/sqs.ts:56]: `~` (Nit). I might just ternary this:

```
const principalOps = this.queue.encryptionMasterKey ? {} : ...
```

File: packages/@aws-cdk/aws-events-targets/lib/sqs.ts [hunk 2]
```diff
@@ -52,14 +52,18 @@ export class SqsQueue implements events.IRuleTarget {
...
+    if (this.queue.encryptionMasterKey == null) {
```
Comment (njlynch) [inline packages/@aws-cdk/aws-events-targets/lib/sqs.ts:57]: More idiomatic in our codebase would be `encryptionMasterKey === undefined` (or even better, `!this.queue. encryptionMasterKey`). Good primer on null vs undefined in Java/TypeScript: https://codeburst.io/javascript-null-vs-undefined-20f955215a2

**Review by njlynch — COMMENTED** [review]
Looks good, with one nitpick and one kinda nitpick.

File: packages/@aws-cdk/aws-events-targets/lib/sqs.ts (hunk 1 above)
Comment (madeline-k) [reply packages/@aws-cdk/aws-events-targets/lib/sqs.ts]: Good idea! updated to incorporate this.

File: packages/@aws-cdk/aws-events-targets/lib/sqs.ts (hunk 2 above)
Comment (madeline-k) [reply packages/@aws-cdk/aws-events-targets/lib/sqs.ts]: I opted to use the ternary operator. But thanks for this info, good to know for future reference!

**Review by njlynch — APPROVED** [review]
🎉  Looks good to me!

**Pull request merged**